src/main/java/com/gard/testbed/activitySystem/IBlackboard.java

```diff
@@ -3,7 +3,9 @@
 /**
  * Created by va922kg on 3/15/16.
  */
-public interface IBlackboard {
+public interface IBlackboard extends IObservable{
+    String getId();
```

Comment:
Why are you mixing the two interfaces? Every board according to the model will be observable. You are binding the interaction you decided on your system with the datarepresentation. Never do that. Interfaces shall be separated